api/tests/integration/domain/services/smart-random/smart-random_test.js

```diff
@@ -661,20 +738,65 @@ describe('Integration | Domain | Stategies | SmartRandom', () => {
           })
         ];
 
-        sinon.stub(_, 'sample').callsFake(function fakeFn(array) {
-          if (array.includes(challengeUrl_2)) {
-            return challengeUrl_2;
-          }
-          return array[0];
-        });
-
         // when
-        const { nextChallenge } = SmartRandom.getNextChallenge({ targetSkills, challenges, knowledgeElements, lastAnswer });
+        const { nextChallenge } = SmartRandom.getNextChallenge({
+          targetSkills,
+          challenges,
+          knowledgeElements,
+          lastAnswer,
+          assessmentId
+        });
 
         // then
         expect(nextChallenge).to.deep.equal(challengeWeb_2_3);
       });
     });
 
+    context('when there is multiply skills and challenge', () => {
+      it('should always return the same challenge at every call', () => {
+        targetSkills = [web2, url2, cnil2];
+        challenges = [challengeWeb_2, challengeWeb_2_bis, challengeUrl_2, challengeCnil_2];
+        lastAnswer = [];
+        knowledgeElements = [];
+
+        for (let i = 0; i < 10;i++) {
+          // when
+          const { nextChallenge } = SmartRandom.getNextChallenge({
+            targetSkills,
+            challenges,
+            knowledgeElements,
+            lastAnswer,
+            assessmentId
+          });
+
+          // then
+          expect(nextChallenge).to.deep.equal(challengeUrl_2);
+        }
+
+      });
+
+      it('should aways return the same challenge at every call', () => {
```

Comment:
Quelle est l'intention de ce test ? Il s'appelle comme le précédant

Reply:
Vérifié que lorsqu'on a deux choix possible, il n'y a pas d'aléatoire et qu'on rechoisit bien la même épreuve.

Reply:
`should still return the same result at every call for another challenge` du coup peut-être ?